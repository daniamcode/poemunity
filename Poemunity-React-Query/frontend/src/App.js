import React, { useState } from 'react'
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'
import './App.scss'
import Dashboard from './components/Dashboard'
import Detail from './components/Detail'
import Header from './components/Header'
import Profile from './components/Profile'
import Login from './components/Login'
import Register from './components/Register'
import PageNotFound from './components/PageNotFound'

export const AppContext = React.createContext();

function App (props) {
  const [contextState, setContextState] = useState({
    elementToEdit: '',
    user: null,
    userId: '',
    username: '',
    config: {},
    adminId: process.env.NODE_ENV === 'development' 
      ? process.env.REACT_APP_ADMIN_PRE
      : process.env.REACT_APP_ADMIN,
    setState(data) {
        const { setState, ...res } = data;
        const newState = { ...contextState, ...res };
        setContextState(newState);
    },
  });
  return (
    <AppContext.Provider value={contextState}>
      <Router>
        <div className='container'>
          <Header />

          <div className='margin-body'>
            <Switch>
              <Route path='/' exact component={Dashboard} />
              <Route path='/profile' component={Profile} />
              <Route path='/login' component={Login} />
              <Route path='/register' component={Register} />
              <Route path='/:genre' exact component={Dashboard} />
              <Route path='/detail/:poemId' exact component={Detail} />
              <Route component={PageNotFound} />
            </Switch>
          </div>
        </div>
      </Router>
    </AppContext.Provider>

  )
}


export default App
