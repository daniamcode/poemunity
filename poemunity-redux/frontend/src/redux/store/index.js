import { applyMiddleware, createStore } from 'redux';
import { createLogger }                 from 'redux-logger';
import thunk                            from 'redux-thunk';
import { composeWithDevTools }          from 'redux-devtools-extension';

import rootReducer      from '../reducers';

const { NODE_ENV } = process.env


// eslint-disable-next-line import/no-mutable-exports
let configureStore;
if (NODE_ENV !== 'production') {
    const enhancer = composeWithDevTools(
        applyMiddleware(thunk, createLogger()),
    );

    // eslint-disable-next-line no-shadow
    configureStore = function configureStore(initialState) {
        return createStore(rootReducer, initialState, enhancer);
    };
} else {
    // eslint-disable-next-line no-shadow
    configureStore = function configureStore(initialState) {
        return createStore(rootReducer, initialState, applyMiddleware(thunk));
    };
}

export default configureStore;
