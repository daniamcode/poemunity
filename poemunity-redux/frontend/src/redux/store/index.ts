import { configureStore } from '@reduxjs/toolkit'
import {rootReducer}      from '../reducers/rootReducer';
import { useDispatch } from 'react-redux'


const store = configureStore({
    reducer: rootReducer,
  })

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export const useAppDispatch: () => AppDispatch = useDispatch // Export a hook that can be reused to resolve types
export default store


// import { applyMiddleware, createStore } from 'redux';
// import { createLogger }                 from 'redux-logger';
// import thunk                            from 'redux-thunk';
// import { composeWithDevTools }          from 'redux-devtools-extension';

// const { NODE_ENV } = process.env

// let configureStore
// if (NODE_ENV !== 'production') {
//     const enhancer = composeWithDevTools(
//         applyMiddleware(thunk, createLogger()),
//     );

//     configureStore = function configureStore(initialState) {
//         return createStore(rootReducer, initialState, enhancer);
//     };
// } else {
//     configureStore = function configureStore(initialState) {
//         return createStore(rootReducer, initialState, applyMiddleware(thunk));
//     };
// }

// export default configureStore;

