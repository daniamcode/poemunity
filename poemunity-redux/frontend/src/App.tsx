import React, { useState } from 'react'
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'
import './App.scss'
import Dashboard from './components/Dashboard'
import Detail from './components/Detail'
import Header from './components/Header'
import Profile from './components/Profile'
import Login from './components/Login'
import Register from './components/Register'
import PageNotFound from './components/PageNotFound'
import { Context } from './typescript/interfaces'

export const AppContext = React.createContext<Context>({
    elementToEdit: '',
    user: '',
    userId: '',
    username: '',
    picture: '',
    config: {},
    adminId: '',
    setState: () => {},
})

function App () {
  const [contextState, setContextState] = useState({
    elementToEdit: '',
    user: '',
    userId: '',
    username: '',
    picture: '',
    config: {},
    adminId: process.env.REACT_APP_ADMIN ?? '',
    setState(data: Context) {
      const { setState, ...res } = data;
      const newState = { ...contextState, ...res };
      setContextState(newState);
    },
  });

  return (
    <AppContext.Provider value={contextState}>
      <Router>
        <div className='container'>
          <Header />

          <div className='margin-body'>
            <Switch>
              <Route path='/' exact component={Dashboard} />
              <Route path='/profile' component={Profile} />
              <Route path='/login' component={Login} />
              <Route path='/register' component={Register} />
              <Route path='/:genre' exact component={Dashboard} />
              <Route path='/detail/:poemId' exact component={Detail} />
              <Route component={PageNotFound} />
            </Switch>
          </div>
        </div>
      </Router>
    </AppContext.Provider>

  )
}


export default App
