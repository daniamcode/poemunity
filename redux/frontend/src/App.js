import React from 'react'
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'
import './App.scss'
import Dashboard from './components/Dashboard'
import Detail from './components/Detail'
import Header from './components/Header'
import Profile from './components/Profile'
import PageNotFound from './components/PageNotFound'

function App (props) {
  return (
    <Router>
      <div className='container'>
        <Header />

        <div className='margin-body'>
          <Switch>
            <Route path='/' exact component={Dashboard} />
            <Route path='/perfil' component={Profile} />
            <Route path='/:genre' exact component={Dashboard} />
            <Route path='/detalle/:poemId' exact component={Detail} />
            <Route component={PageNotFound} />
          </Switch>
        </div>
      </div>
    </Router>
  )
}


export default App
