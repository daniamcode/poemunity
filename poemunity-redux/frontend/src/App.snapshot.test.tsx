import { render } from '@testing-library/react';
import App from './App'
import {
  QueryClient,
  QueryClientProvider,
} from "react-query"
import { Provider } from 'react-redux';
import store from './redux/store';


describe('App snapshot', () => {
  const queryClient = new QueryClient();

  const tree = render(
    <Provider store={store}>
    <QueryClientProvider client={queryClient}>
      <App />
    </QueryClientProvider>
    </Provider>
  )

  test('should match', () => {
    expect(tree.asFragment()).toMatchSnapshot()
  })
})
