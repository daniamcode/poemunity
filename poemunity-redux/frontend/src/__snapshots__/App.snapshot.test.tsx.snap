// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <section
      class="header"
    >
      <div
        class="header__dropdown"
      >
        <div
          class="MuiPaper-root MuiAccordion-root accordion MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            aria-controls="panel1a-content"
            aria-disabled="false"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root"
            id="panel1a-header"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content"
            >
              <div
                class="header__dropdown-categories-icon"
              />
              <p
                class="header__dropdown-categories"
              >
                Categories
              </p>
            </div>
            <div
              aria-disabled="false"
              aria-hidden="true"
              class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <div
                  aria-labelledby="panel1a-header"
                  id="panel1a-content"
                  role="region"
                >
                  <div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/anniversary"
                      >
                        Anniversary
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/friendship"
                      >
                        Friendship
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/funny"
                      >
                        Funny
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/haikus"
                      >
                        Haikus
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/hope"
                      >
                        Hope
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/life"
                      >
                        Life
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/love"
                      >
                        Love
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/sad"
                      >
                        Sad
                      </a>
                    </div>
                  </div>
                  <div
                    class="MuiAccordionDetails-root"
                  >
                    <a
                      class="header__dropdown-subcategories"
                      href="/"
                    >
                      All
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="header__logo"
      >
        <a
          class="header__text-logo-first"
          href="/"
        >
          P
        </a>
        <a
          class="header__logo-icon"
          href="/"
        />
        <a
          class="header__text-logo-second"
          href="/"
        >
          emunity
        </a>
      </div>
      <p
        class="list__presentation"
      >
        Your poem community!!!
      </p>
      <div
        class="separator"
      />
      <button
        class="header__login"
      />
    </section>
    <div
      class="margin-body"
    >
      <main
        class="dashboard"
      >
        <div
          class="dashboard__accordion"
        >
          <div
            class="MuiPaper-root MuiAccordion-root accordion MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              aria-controls="panel1a-content"
              aria-disabled="false"
              aria-expanded="false"
              class="MuiButtonBase-root MuiAccordionSummary-root"
              id="panel1a-header"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiAccordionSummary-content"
              >
                <div
                  class="header__dropdown-categories-icon"
                />
                <p
                  class="header__dropdown-categories"
                >
                  Categories
                </p>
              </div>
              <div
                aria-disabled="false"
                aria-hidden="true"
                class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-hidden"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby="panel1a-header"
                    id="panel1a-content"
                    role="region"
                  >
                    <div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/anniversary"
                        >
                          Anniversary
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/friendship"
                        >
                          Friendship
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/funny"
                        >
                          Funny
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/haikus"
                        >
                          Haikus
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/hope"
                        >
                          Hope
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/life"
                        >
                          Life
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/love"
                        >
                          Love
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/sad"
                        >
                          Sad
                        </a>
                      </div>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/"
                      >
                        All
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="dashboard__list"
        >
          <div
            class="makeStyles-root-2"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="dashboard__ranking"
        >
          <div
            class="makeStyles-root-2"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
